### PR TITLE
Add pagination

### PR DESCRIPTION
**Pull request by munisystem**
## WHY
To support pagination.

## WHAT
- Add template and generate it when ``apig new`` executed.
- Add pagination rogic to generated controllers.

### CODE REVIEW

File: _templates/controller.go.tmpl [hunk 1]
```diff
@@ -17,10 +17,31 @@ func Get{{ pluralize .Model.Name }}(c *gin.Context) {
...
+	err = db.Select(fields).Find(&{{ pluralize (tolower .Model.Name) }}).Error
+
+	if err != nil {
+		c.JSON(500, gin.H{"error": "error occured"})
+		return
+	}
+
+	// paging
+	var index int
+	if len(users) < 1 {
```
Comment (dtan4) [inline _templates/controller.go.tmpl]: Please replace `users` to `{{ pluralize (tolower .Model.Name) }}`

File: _templates/controller.go.tmpl [hunk 2]
```diff
@@ -17,10 +17,31 @@ func Get{{ pluralize .Model.Name }}(c *gin.Context) {
...
+		c.JSON(500, gin.H{"error": "error occured"})
+		return
+	}
+
+	// paging
+	var index int
+	if len({{ .pluralize (tolower .Model.Name) }}) < 1 {
+		index = 0
+	} else {
+		index = int({{ .pluralize (tolower .Model.Name) }}[len({{ .pluralize (tolower .Model.Name) }})-1].ID)
```
Comment (dtan4) [inline _templates/controller.go.tmpl]: `pluralize` is function, so please remove `.` .

Comment (munisystem) [inline _templates/controller.go.tmpl]: Oh...Sorry...

**Comment by munisystem** [comment]
After `gen` subcommand executed, `controllers/user.go` is as fallows.
```
	// paging
	var index int
	if len(users) < 1 {
		index = 0
	} else {
		index = int(users[len(users)-1].ID)
	}
	pagination.SetHeaderLink(c, index)
```

**Comment by dtan4** [comment]
LGTM 👍 Thanks ❗

**Pull request merged**